test/validation/api/timer/timer.c

```diff
@@ -755,11 +755,20 @@ static void timer_single_shot(odp_queue_type_t queue_type, odp_timer_tick_type_t
 	CU_ASSERT_FATAL(ret == 0);
 	CU_ASSERT_FATAL(capa.max_tmo.max_tmo > 0);
 
+	/* Use timeout and resolution values that are within capability limits */
 	if (capa.max_tmo.max_tmo < tmo_ns) {
 		tmo_ns = capa.max_tmo.max_tmo;
 		res_ns = capa.max_tmo.res_ns;
 	}
 
+	if (capa.max_res.res_ns > res_ns) {
+		tmo_ns = capa.max_res.max_tmo;
+		res_ns = capa.max_res.res_ns;
+	}
+
+	/* Test expects better resolution than 0.5x timeout */
+	CU_ASSERT_FATAL(res_ns < tmo_ns / 2);
```

Comment:
Should this be tried with the other capability as well in case of failure, i.e. fall back to try with `max_tmo` values if `max_res` values don't comply?

Reply:
Fixed in v2. Using odp_timer_res_capability() for limiting res_ns and min_tmo values.